### PR TITLE
Add meta update object to SC offchain updates

### PR DESCRIPTION
**Pull request by uwiger**
See [PT #167808073](https://www.pivotaltracker.com/story/show/167808073)

This PR adds a `{meta, binary()}` update op to the channel_offchain_updates. This could be added to any update request, but is currently only supported by the fsm for `upd_transfer`. The `fsm_SUITE` has been slightly extended to verify that meta objects can be added, but the WS API has not yet been changed to support them.

This PR depends on a PR to aeserialization.

### CODE REVIEW

**Comment by lucafavatella** [comment]
Kept WIP to label only - for avoiding duplication.

File: apps/aechannel/src/aesc_offchain_update.erl [hunk 1]
```diff
@@ -102,6 +112,7 @@ from_db_format(Tuple) ->
         #deposit{} -> Updated;
         #create_contract{} -> Updated;
         #call_contract{} -> Updated;
+        #meta{} -> Updated;
```
Comment (velzevur) [inline apps/aechannel/src/aesc_offchain_update.erl]:
```suggestion
```
This is not reachable

File: apps/aechannel/src/aesc_fsm.erl [hunk 2]
```diff
@@ -259,11 +260,14 @@ upd_deposit(Fsm, #{amount := Amt} = Opts) when is_integer(Amt) ->
     lager:debug("upd_deposit(~p)", [Opts]),
     gen_statem:call(Fsm, {upd_deposit, Opts}).
 
-upd_transfer(_Fsm, _From, _To, Amount) when Amount < 0 ->
+upd_transfer(Fsm, From, To, Amount) ->
+    upd_transfer(Fsm, From, To, Amount, #{}).
+
+upd_transfer(_Fsm, _From, _To, Amount, _Opts) when Amount < 0 ->
```
Comment (tolbrino) [inline apps/aechannel/src/aesc_fsm.erl]: Possibly unrelated: Why would a transfer of 0 make sense?

File: rebar.config [hunk 3]
```diff
@@ -68,7 +68,7 @@
         {aebytecode, {git, "https://github.com/aeternity/aebytecode.git", {ref, "10cc127"}}},
 
         {aeserialization, {git, "https://github.com/aeternity/aeserialization.git",
-                          {ref,"4a07297"}}},
+                          {ref,"d87adfd"}}},
```
Comment (tolbrino) [inline rebar.config]: Note: Requires https://github.com/aeternity/aeserialization/pull/9 to be merged first.

File: rebar.lock [hunk 4]
```diff
@@ -18,7 +18,7 @@
   0},
  {<<"aeserialization">>,
   {git,"https://github.com/aeternity/aeserialization.git",
-      {ref,"4a072977745571ceb6cf34e6387a96b808ae20b3"}},
+      {ref,"d87adfd58333b70a725e707a2caf8a4604e6a126"}},
```
Comment (tolbrino) [inline rebar.lock]: Note: Requires aeternity/aeserialization#9 to be merged first.

File: apps/aechannel/src/aesc_fsm.erl (hunk 2 above)
Comment (velzevur) [reply apps/aechannel/src/aesc_fsm.erl]: This is a funny edge case. I don't see a reason of forbidding it, nor do I see any meaningful use of it.

Comment (uwiger) [reply apps/aechannel/src/aesc_fsm.erl]: Well, it doesn't break anything, so why would we disallow it? ;-)

File: apps/aechannel/src/aesc_offchain_update.erl [hunk 5]
```diff
@@ -320,62 +339,70 @@ fields2update(call_contract, [ {caller, CallerId},
...
     [ {caller,      id},
       {contract,    id},
       {abi_version, int},
       {amount,      int},
       {gas,         int},
       {gas_price,   int},
       {call_data,   binary},
-      {call_stack,  [int]}].
+      {call_stack,  [int]}];
+update_serialization_template(?UPDATE_VSN, meta) ->
```
Comment (velzevur) [inline apps/aechannel/src/aesc_offchain_update.erl]: I see this is allowed to be deserialized only using the new version. OK.

**Comment by uwiger** [comment]
Updated deps in line with @tolbrino's comments.

File: apps/aechannel/src/aesc_offchain_update.erl [hunk 6]
```diff
@@ -1,6 +1,7 @@
 -module(aesc_offchain_update).
 
--define(UPDATE_VSN, 1).
+-define(UPDATE_VSN, 2).
```
Comment (velzevur) [inline apps/aechannel/src/aesc_offchain_update.erl]: This makes [serialization](https://github.com/aeternity/aeternity/blob/2db747d7e378d4bf5235e532edc5a75ae150f0ed/apps/aechannel/src/aesc_offchain_update.erl#L265) to use v2 off-chain serialization. This is good but being not backwards compatible regarding the off-chain protocol communication, I expect [some system tests](https://github.com/aeternity/aeternity/blob/master/system_test/common/aest_channels_SUITE.erl#L16-L19) to fail

**Pull request merged**